internal/service/vpc/resource_azure_vpc_peering_connection.go

```diff
@@ -0,0 +1,363 @@
+	},
+	"state": {
+		Computed:    true,
+		Type:        schema.TypeString,
+		Description: "State of the peering connection",
+	},
+	"state_info": {
+		Computed:    true,
+		Type:        schema.TypeMap,
+		Description: "State-specific help or error information",
+	},
+	"peering_connection_id": {
+		Computed:    true,
+		Type:        schema.TypeString,
+		Description: "Cloud provider identifier for the peering connection if available",
+	},
+	"peer_azure_app_id": {
+		Required:    true,
+		ForceNew:    true,
+		Type:        schema.TypeString,
+		Description: schemautil.Complex("Azure app registration id in UUID4 form that is allowed to create a peering to the peer vnet").ForceNew().Build(),
+	},
+	"peer_azure_tenant_id": {
+		Required:    true,
+		ForceNew:    true,
+		Type:        schema.TypeString,
+		Description: schemautil.Complex("Azure tenant id in UUID4 form.").ForceNew().Build(),
+	},
+}
+
+func ResourceAzureVPCPeeringConnection() *schema.Resource {
+	return &schema.Resource{
+		Description:   "The Azure VPC Peering Connection resource allows the creation and management of Aiven VPC Peering Connections.",
+		CreateContext: resourceAzureVPCPeeringConnectionCreate,
+		ReadContext:   resourceAzureVPCPeeringConnectionRead,
+		DeleteContext: resourceAzureVPCPeeringConnectionDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceAzureVPCPeeringConnectionImport,
+		},
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(2 * time.Minute),
+			Delete: schema.DefaultTimeout(2 * time.Minute),
+		},
+
+		Schema: aivenAzureVPCPeeringConnectionSchema,
+	}
+}
+
+func resourceAzureVPCPeeringConnectionCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	var (
+		pc     *aiven.VPCPeeringConnection
+		err    error
+		region *string
+	)
+
+	client := m.(*aiven.Client)
+	projectName, vpcID := schemautil.SplitResourceID2(d.Get("vpc_id").(string))
+	if projectName == "" || vpcID == "" {
+		return diag.Errorf("incorrect VPC ID, expected structure <PROJECT_NAME>/<VPC_ID>")
+	}
+
+	azureSubscriptionId := d.Get("azure_subscription_id").(string)
+	vnetName := d.Get("vnet_name").(string)
+
+	// Azure related fields are only available for VPC Peering Connection resource but
+	// not for Transit Gateway VPC Attachment therefore ResourceData.Get retrieves nil
+	// for fields that are not present in the schema.
+	var peerAzureAppID, peerAzureTenantID, peerResourceGroup string
+	if v, ok := d.GetOk("peer_azure_app_id"); ok {
+		peerAzureAppID = v.(string)
+	}
+	if v, ok := d.GetOk("peer_azure_tenant_id"); ok {
+		peerAzureTenantID = v.(string)
+	}
+	if v, ok := d.GetOk("peer_resource_group"); ok {
+		peerResourceGroup = v.(string)
+	}
+
+	if _, err = client.VPCPeeringConnections.Create(
+		projectName,
+		vpcID,
+		aiven.CreateVPCPeeringConnectionRequest{
+			PeerCloudAccount:  azureSubscriptionId,
+			PeerVPC:           vnetName,
+			PeerRegion:        region,
```

Comment:
It seems `region` is always `nil`. Is it expected?

Reply:
:+1: